addons/mail/models/res_partner.py

```diff
@@ -35,7 +36,19 @@ def _compute_contact_address_inline(self):
             partner.contact_address_inline = re.sub(r'\n(\s|\n)*', ', ', partner.contact_address).strip().strip(',')
 
     def _compute_im_status(self):
-        super()._compute_im_status()
+        status_by_partner = {}
+        for presence in self.env["mail.presence"].search([("user_id", "in", self.user_ids.ids)]):
```

Comment:
isn't it somehow a related on user_ids then ? or a compute based on each user_id.im_status ?

Reply:
`user_ids.im_status` is overridden to add custom statuses such as `leave_online`, `presence_{work_location}_online`, and similar. I feel like using presence is the easiest way to compute these statuses.